tests/test_metric_output.py

```diff
@@ -0,0 +1,50 @@
+import requests
+import yaml
+
+
+class TestMetricOutput:
+    def test_prometheus_server_reply(self, setup_vcsa_url):
+        response = requests.get(setup_vcsa_url)
+        metrics = self.process_response_data(response)
+
+        with open('mockingServer/data/vcsa_output_comparison.yaml', 'r') as fp:
+            comparison_data = yaml.safe_load(fp)
+
+        for key in comparison_data.keys():
+            difference = set(comparison_data[key]).difference(metrics)
+            assert len(difference) == 0, 'No difference between the metric dump and the Prometheus reply expected.'
+
+    def test_backend_recovery_from_down(self, request, setup_vcsa_url, setup_vcenter):
+        # Backend up test
+        self.backend_up(setup_vcsa_url)
+        # Backend down test
+        setup_vcenter.name = 'foobar-vcenter.whatever.domain'
+        response = requests.get(setup_vcsa_url)
+        metrics = self.process_response_data(response)
+        setup_vcenter.name = request.config.getoption('--host')
+        assert len(metrics) == 0, 'If the backend is down, no vcsa service metrics are expected'
+        # Backend up test
+        self.backend_up(setup_vcsa_url)
```

Comment:
don't we need another assert, to ensure we are up again?

Reply:
The assert is happening in the backend_up helper method for this testcase, reference to line 40 in this code. 'test_backend_recovery_from_down' calls 'beckend_up' which includes the assert. I can also return metrics and implement assert into this testcase. I think this would be more clear.
What do you think? :)

Reply:
Changed it, I also changed the message at the second backend_up test.